assets/training/model_management/src/run_model_preprocess.py

```diff
@@ -131,7 +133,18 @@ def run():
             preprocess_args.update(download_details.get("properties", {}))
             preprocess_args["misc"] = download_details.get("misc", [])
 
-    preprocess_args["task"] = task_name if task_name else preprocess_args.get("task")
+    if task_name is None or not SupportedTasks.has_value(task_name):
+        task_name = preprocess_args.get("task")
```

Comment:
(nit) lower?

Reply:
Also, I remember we seeing an isue where "text-generation", "text-generation-inference" or something like that are recognized as tasks. Could we also have such mappings as well in case the task name used is different but the actual functionality is the same?

Reply:
Handled case sensitivity. 
text-generation-inference is not supported. in this case it will be fetched from HFAPI.